enhancements/monitoring/alerting-consistency.md

```diff
@@ -0,0 +1,195 @@
+Critical Alerts should be reserved for situations that jeopardize the health
+if not immediately addressed or if the cluster is already in a critical state.
+
+Some critical states are:
+* loss or impending loss of etcd-quorum.
+* inability to route application traffic externally or internally
+(dataplane disruption)
+* inability to start any new pod other than capacity.  EG, if the API server
+were to restart or need to be rescheduled, it would not able to start on another
+host
+
+In otherwords, critical alerts are something that require someone to get out
+of bed in the middle of the night and fix something **right now** or they will
+be faced with a disaster.
+
+An example of something that is NOT a critical alert:
+* MCO and/or related components are completely dead/crash looping.
+
+Even though the above is quite significant, there is no risk of immediate loss
+of the cluster.
+
+You might be thinking to yourself: "But my component is **super** important, and
+if it's not working, the user can't do X, which user definitely will want."  All
+that is probably true, but that doesn't make your alert "critical" it just makes
+it worthy of an alert.
+
+### Warning Alerts
+
+TL/DR: fix this soon, or some things won't work and upgrades will probably be
+blocked.
+
+If your alert does not meet the criteria in "Critical Alerts" above, it belongs
+to the warning level or lower.
+
+Most alerts probably fit in this category.  This level should represent degraded
+or unavailable services that do not jeopardize the immediate life of cluster.
+
+Timeline:  ~60 minutes
+
+60 minutes?  That seems high!  That's because it is high.  We want to reduce
+the noise.  We've done a lot to make clusters and operators auto-heal
+themselves.  The whole idea is that if a condition has persisted for more than
+60 minutes, it's unlikely to be resolved without intervention any time later.
+
+#### Example 1: All machine-config-daemons are crashlooping.
+
+That seems significant
+and it probably is, but nobody has to get out of bed to fix this **right now**.
+But, what if a machine dies, I won't be able to get a replacement?  Yeah, that
+is probably true, but how often does that happen?  Also, that's an entirely
+separate set of concerns.
+
+#### Example 2:  I have an operator that needs at least 3 of something to work.
+* With 2/3 replicas, warning alert after 60M.
+* With 1/3 replicas, warning alert after 10M.
+* With 0/3 replicas, warning alert after 5M.
+
+Q: How long should we wait until the above conditions rises to a 'critical'
+alert?
+
+A: It should never rise to the level of a critical alert, it's not critical.  If
+it was, this section would not apply.
+
+#### Example 3:  I have an operator that only needs 1 replica to function
+
+Well, if the cluster can upgrade with only 1 replica, and the the service is
+not unavailable despite other replicas being unavailable, this can probably
+be just an info-level alert.  We should have a "minor" alert, TBD.  But for now,
```

Comment:
What's the advantage of `Minor` over `info` ?

Reply:
Info doesn't imply problem.  Nobody wants an 'info' alert.

Reply:
Seems far fetched to assume your opinion is the same as everyone's

Reply:
Someone else shared this: https://docs.openshift.com/container-platform/4.6/monitoring/managing-alerts.html#searching-alerts-silences-and-alerting-rules_managing-alerts

> Info. The alert is provided for informational purposes only.

So, even by our own docs, "Info" != problem.  Here's an example of an 'info' alert: "ClusterUpdateAvailable".  Is that a problem?  Should we put problems in the same category as that?  I don't think so.

Reply:
`ClusterUpdateAvailable` is an `info` alert today, because there is no other mechanism for getting a recommendation out to cluster admins.  We have [an internal ticket][1] (sorry non-Red-Hat folks) talking about a potential `Recommendation` CRD, and if/when we get something like that, the CVO will definitely drop the `ClusterUpdateAvailable` alert and make it a `Recommendation`.

[1]: https://issues.redhat.com/browse/WRKLDS-186

Reply:
OSD SRE [routes only CRITICAL and WARNING alerts out of the clusters](https://github.com/openshift/configure-alertmanager-operator/blob/master/pkg/controller/secret/secret_controller.go#L137) and pages only on the former. INFOs are only available if an SRE directly checks the alerts on a cluster.

From that perspective anything that's indicative of a lower–impact issue needs to be a WARNING – lest it gets completely missed – and anything that doesn't should be reduced to INFO to decrease noisiness.